### PR TITLE
Make generate_testdata_advanced work again

### PR DESCRIPTION
**Pull request by cfra**
Making my first steps with Juntagrico and toying around with things, I came across the `generate_testdata_advanced` management command.

This management sounded like it might be interesting so get a first understanding of the data model.

Unfortunately, it seems that some updates to the models broke it.

Also, this script relied on various external services that were not available anymore.

I took the time to clean it up a bit and to replace usage of those external services by usage of Faker.

I also added it to the testsuite, so any future breakage will hopefully not go unnoticed.

All in all I am not sure how useful this management command is in the end though.

If you feel it is not that useful and it would be better to just remove it, let me know and I will update the PR. :slightly_smiling_face:

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -2,6 +2,7 @@ Django==3.1.5
 django-crispy-forms==1.10.0
 django-impersonate==1.7.2
 django-polymorphic==3.0.0
+faker==5.8.0
```
Comment (motu55) [inline requirements.txt:5]: this way the dependency will also be installed on produktive systems which makes no sense and just uses up valluable space.

**Review by motu55 — CHANGES_REQUESTED** [review]
there is a method for quick testdata generation which is in most cases sufficient
that is also the reason why it has been neglected for so long
we can continue to use this but it would require a mechanism so that it will not be part of a release and the fake requirement is not installed on a production environment

**Comment by cfra** [comment]
I totally agree the list of production requirements should be kept as small as possible. :+1:

I just did not really know where to put the `faker` requirement, because I did not find a place for dev dependencies.

This might be related to the fact that I did not find any information how a development environment for Juntagrico is usually set up.

I just installed the requirements into a venv and then used `./manage.py runserver`, but I am not sure if using the default SQLite database is the recommended way.

Is there any particular approach you would suggest for this?

I could imagine using something like `requirements-dev.txt`, but nowadays I mostly use Poetry, so I don't know what the best common practice for the `requirements.txt` approach is.

**Comment by pascalfree** [comment]
Thanks for your contribution.
You did the right development setup. Nothing fancy here. :-)

I agree with the `requirements-dev.txt` idea.

Also `manage.py runserver` seems to work fine even if faker is not installed. So no need to change anything else.

I have not heard of poetry before. Looks interesting. Would you recommend it for a project like this one?

**Comment by motu55** [comment]
i would also for the moment go for a requirements-dev.txt file
also make sure not only the runserver works but alos the gunicorn startup works since it is the one used in productive systems.

i also think that poetry look interesting. but this is out of the skope of this pullrequest. @pascalfree we could create an investigative issue where we have a discussion about that.

**Comment by cfra** [comment]
> I agree with the `requirements-dev.txt` idea.
> 
> Also `manage.py runserver` seems to work fine even if faker is not installed. So no need to change anything else.

I am not sure if this is completely true.

As the script is now executed as part of the test suite, the development requirements also need to be available on CI.

> I have not heard of poetry before. Looks interesting. Would you recommend it for a project like this one?

I don't think I have sufficient experience with a Django application like Juntagrico that is provided as a package on PyPI, to provide a good recommendation.

For the Django projects I have been working on in commercial contexts, using Poetry was very nice because it has a better solver for complex dependency situations and also made things a lot more reproducible by being able to maintain a list of versions for all packages installed.

I believe @motu55 's suggestion to maybe discuss this as a separate issue is probably best, as this really is a different topic. :slightly_smiling_face:

**Comment by cfra** [comment]
> > I agree with the `requirements-dev.txt` idea.
> > Also `manage.py runserver` seems to work fine even if faker is not installed. So no need to change anything else.
> 
> I am not sure if this is completely true.
> 
> As the script is now executed as part of the test suite, the development requirements also need to be available on CI.

Seems like this is indeed the case.

Looking at `.travis.yml` it seems there are already `coverage` and `django_coverage_plugin` being installed as development dependencies.

So maybe something like this would make sense?

```diff
diff --git a/.travis.yml b/.travis.yml
index bca277be..cb333c90 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -4,7 +4,7 @@ python:
   - '3.8'
 install:
   - pip install --upgrade -r requirements.txt
-  - pip install coverage django_coverage_plugin
+  - pip install --upgrade -r requirements-dev.txt
 before_script:
   - curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./cc-test-reporter
   - chmod +x ./cc-test-reporter
diff --git a/requirements-dev.txt b/requirements-dev.txt
index 930eb814..9fca8f99 100755
--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -1 +1,3 @@
+coverage==5.4
+django-coverage-plugin==1.8.0
 faker==5.8.0
```

Let me know what you think.

(And sorry this is getting so complicated :grimacing: )

**Comment by pascalfree** [comment]
I agree with that solution. Makes it easier to add dev requirements in the future.

**Pull request merged**